src/main/kotlin/kscript/app/creator/PackageCreator.kt

```diff
@@ -6,28 +6,51 @@ import kscript.app.model.Script
 import kscript.app.util.Executor
 import kscript.app.util.FileUtils
 import kscript.app.util.Logger.infoMsg
+import kscript.app.util.ScriptUtils.dropExtension
 import java.nio.file.Path
+import java.nio.file.Paths
+import java.io.File
 
 class PackageCreator(private val executor: Executor) {
     /**
      * Create and use a temporary gradle project to package the compiled script using capsule.
      * See https://github.com/puniverse/capsule
      */
     fun packageKscript(basePath: Path, script: Script, jarArtifact: JarArtifact): Path {
+        val baseName = script.scriptName.dropExtension()
+
         infoMsg("Packaging script '${script.scriptName}' into standalone executable...")
 
         // create exec_header to allow for direction execution (see http://www.capsule.io/user-guide/#really-executable-capsules)
         // from https://github.com/puniverse/capsule/blob/master/capsule-util/src/main/resources/capsule/execheader.sh
-        FileUtils.createFile(basePath.resolve("exec_header.sh"), Templates.executeHeader)
+        var hdr = Templates.executeHeader
+        val parts = mutableListOf("")
+
+        for (opt in script.kotlinOpts) {
+            val s = opt.value
+            if (s.startsWith("-J")) {
+                parts.add(s.substring(2))
+            }
+        }
+        val opts = parts.joinToString(" ").trim()
+        hdr = hdr.replace("java -jar", "java $opts -jar")
```

Comment:
It will be better to create a function for the generation of exectuteHeader and pass $opts as an argument to this function than to replace the value textually.

Reply:
Not sure _exactly_ what you mean, but I'll have a go.

Reply:
Done.